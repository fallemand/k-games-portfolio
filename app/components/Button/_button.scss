@import '../../styles/variables';

$button-green-color: #00b67d;
$button-default-color: #666666;
$button-blue-color: #1b3bd4;
$button-gradient: linear-gradient(180deg, hsla(0, 0%, 100%, 0.3), hsla(0, 0%, 100%, 0));

.button {
  border-radius: 999px;
  box-shadow: inset 0 -4px 0 rgba(0, 0, 0, 0.3), 0 4px 6px rgba(0, 0, 0, 0.22);
  color: #eee;
  font-weight: $font-light;
  font-size: $font-size-14;
  width: 160px;
  margin: 0 5px;
  padding: 8px 0 10px;
  cursor: pointer;

  &.button--green {
    background: $button-green-color $button-gradient;
  }
  &.button--default {
    background: $button-default-color $button-gradient;
  }
  &.button--blue {
    background: $button-blue-color $button-gradient;
  }
}
