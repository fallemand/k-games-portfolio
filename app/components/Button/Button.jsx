import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import './_button.scss';

const Button = ({ className, color, children, onClick, ...props }) => (
  <button
    type="button"
    className={classnames(className, 'button', `button--${color}`)}
    onClick={onClick}
    {...props}
  >
    {children}
  </button>
);

Button.colors = {
  DEFAULT: 'default',
  GREEN: 'green',
  BLUE: 'blue',
};

Button.propTypes = {
  className: PropTypes.string,
  color: PropTypes.oneOf(Object.values(Button.colors)),
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func.isRequired,
};

Button.defaultProps = {
  className: null,
  color: Button.colors.DEFAULT,
};

export default Button;
