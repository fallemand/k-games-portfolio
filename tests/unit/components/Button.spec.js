import React from 'react';
import Button from '../../../app/components/Button';

describe('Button', () => {
  const props = {
    onClick: jest.fn(),
    color: Button.colors.GREEN,
    className: '__CLASSNAME__',
  };

  it('render with props', () => {
    const component = shallow(<Button {...props}>__TEST__</Button>);
    expect(escapeSnapshot(component)).toMatchSnapshot();
  });

  it('should call onClick when on Button click', () => {
    const component = shallow(<Button {...props}>__TEST__</Button>);
    component.find('.button').simulate('click');
    expect(props.onClick).toHaveBeenCalled();
  });
});
