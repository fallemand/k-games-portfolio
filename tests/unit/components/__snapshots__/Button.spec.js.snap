// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render with props 1`] = `
"<button type='button' className='__CLASSNAME__ button button--green' onClick={[Function: mockConstructor]}>
  __TEST__
</button>"
`;
